toolchain/semantics/semantics_node_kind.h

```diff
@@ -16,17 +16,32 @@ CARBON_DEFINE_RAW_ENUM_CLASS(SemanticsNodeKind, uint8_t) {
 #include "toolchain/semantics/semantics_node_kind.def"
 };
 
+// Whether a node is a terminator or part of the terminator sequence. The nodes
+// in a block appear in the order NotTerminator, then TerminatorSequence, then
+// Terminator, which is also the numerical order of these values.
+enum class SemanticsTerminatorKind {
+  // This node is not a terminator.
+  NotTerminator,
+  // This node is not itself a terminator, but forms part of a terminator
+  // sequence.
+  TerminatorSequence,
```

Comment:
Looking at the code, couldn't this just be NotTerminator? I see code that compares with Terminator, but nothing comparing with NotTerminator or TerminatorSequence.

Partly I'm commenting because of the macros: terminators are nodes, so NODE -> TERMINATOR felt a little odd to me. But for one macro, I kind of shrugged. Now it's two, so I feel I should comment; maybe you could do something like `CARBON_SEMANTICS_NODE_KIND_WITH_TERMINATOR_KIND(BranchIf, TerminatorSequence)` and have one macro? It's borderline, but if more details keep getting added, I think it shouldn't be a new macro each time.

Reply:
This is used in `SemanticsIR::Verify`, in `prior_kind > node_kind`, which is checking for the case of `Terminator` followed by `NotTerminator` or `TerminatorSequence`, and for the case of `TerminatorSequence` followed by `NotTerminator`.

I'll merge this and send a separate PR for the suggested macro change.